### PR TITLE
Use expected values to improve strictness computation

### PR DESCRIPTION
**Pull request by neumanrq**
When @carpodaster  and me have proposed the [first version of reviewer strictness computation](https://github.com/rails-girls-summer-of-code/rgsoc-teams/pull/741) I made a mistake because of oversimplification. As @mkalininait correctly pointed out, our current formula gets slightly unfair when the amount of ratings is not equal for all reviewers.

Because I was a bit confused at all I started formalizing our problem en détail [here](https://github.com/neumanrq/fair_reviewers/blob/master/theory/index.pdf), where the new formula is taken from (It is the formula for `S^*`, currently Definiton 9 on page 5).

The short version is: After merging this, the strictness will be computed by determining the expectation value for each reviewer and comparing it to the "overall" computation value, instead of a more simple comparison of "amount of points assigned by reviewer A" with "amount points assigned by an average reviewer". 

I'd love to take @carpodaster into account: Maybe we could pair on this during next week and check it against real data. Of course, everyone is invited to ask sceptical questions on this 👀  😄

### CODE REVIEW

**Comment by mkalininait** [comment]
Related to this issue: https://github.com/rails-girls-summer-of-code/selection_process/issues/60

**Comment by mkalininait** [comment]
Sorry, I closed this PR by accident. Reopened it already.

**Comment by mkalininait** [comment]
Hi @neumanrq First of all, sorry for not showing up for some time. We had an exhausting selection and then were exhaling for a couple of weeks :D 

I've tested this PR on the production data and prepared a spreadsheet where you can compare the OLD corrected average and NEW corrected average. I'll gladly share it with you if you give me your gmail account. Briefly speaking, we have 186 applications; 9 applications have absolute difference between the old and new corrected averages more than 0.5; two versions of top-20 applications (sorted by new and old corrected averages) differ only in 1 application, which is not a lot, but can be significant for those specific people from flesh and blood.

I'm still digging through the theoretical part of the research and will leave my feedback here once I finish.

Thanks again for your work! We really appreciate it!

**Comment by klappradla** [comment]
Thanx a million for your efforts on this @neumanrq 🙏 
And sorry for leaving this lingering around for so long 🙈 

I tbh cannot really say whether or not this fixes the problem we have (plus I don't even recall what the problem was) - but I trust your brain, so I'd just 1, 2, 3 merge this is no one is against it (cc @mkalininait )

**Comment by neumanrq** [comment]
Thanks @klappradla for taking care :) 

Oh yes, I'm completely aware that this is hard to jump in. I'm still convinced these changes will fix the problem described above. I should take the time and do a little write-up to clarify and document the motivation behind these parts for you, the other maintainers, and also future-me 😬

**Comment by klappradla** [comment]
Anyways ❤️💛💚💙💜
